### PR TITLE
Migrate snap to core24 + jazzy

### PR DESCRIPTION
**Pull request by florcabral**

### CODE REVIEW

File: snap/local/localization_launcher.sh [hunk 1]
```diff
@@ -29,4 +29,4 @@ else
 fi
 
 ros2 launch nav2_bringup localization_launch.py \
-  params_file:=${CONFIG_FILE} map:=${MAP_FILE} use_sim_time:=${SIMULATION}
+  params_file:=${CONFIG_FILE} map:=${MAP_FILE}
```
Comment (giusebar) [inline snap/local/localization_launcher.sh]: why the use_sim_time / simulation option has been removed?

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -135,3 +137,14 @@ parts:
     source: snap/local/configuration_templates/
     organize:
       '*.yaml': usr/share/ros2-nav2/configuration_templates/
+  jazzy-desktop-libamdhip-exec-stack:
```
Comment (artivis) [inline snap/snapcraft.yaml:142]: Pedantic: add a space before the new part.

File: snap/local/localization_launcher.sh (hunk 1 above)
Comment (florcabral) [reply snap/local/localization_launcher.sh]: Yes actually we can keep the option on localization and slam launchers, but not navigation (ie, [test job fails](https://github.com/ubuntu-robotics/ros2-nav2-snap/actions/runs/10597697531/job/29369103804) if so).

Comment (giusebar) [reply snap/local/localization_launcher.sh]: the navigation_launcher has the use_sim_time parameter tho https://github.com/ros-navigation/navigation2/blob/jazzy/nav2_bringup/launch/navigation_launch.py#L34 , maybe it is just the amcl use_sim_time parameter in the template configuration that has to be removed?

**Review by giusebar — APPROVED** [review]

**Pull request merged**